js/src/edit-block/components/field-settings.js

```diff
@@ -58,7 +58,7 @@ const FieldSettings = ( {
 
 					const SettingComponent = getSettingsComponent( setting.type );
 					const key = `field-setting-${ index }`;
-					const value = null === field[ setting.name ] ? setting.default : field[ setting.name ];
+					const value = field.hasOwnProperty( setting.name ) ? field[ setting.name ] : setting.default;
```

Comment:
`null === field[ setting.name ]` was the wrong logic. It should have been `undefined === field[ setting.name ]`, but  I think `field.hasOwnProperty( setting.name ) ` is even simpler.